src/uk/gov/hmcts/contino/AzPrivateDns.groovy

```diff
@@ -44,19 +44,29 @@ class AzPrivateDns {
 
         def ttl = this.environmentDnsConfigEntry.ttl
         def zone = this.environmentDnsConfigEntry.zone
+        def aRecordSet
+        def cnameRecordSet
 
+        // if no cname exists in public dns create A record
         this.steps.echo "Registering DNS for ${recordName} to ${serviceIP} with ttl = ${ttl}"
-        def aRecordSet
+        // check for existing record
         try {
           aRecordSet = this.az.az "network private-dns record-set a show -g ${resourceGroup} -z ${zone} -n ${recordName} --subscription ${subscription} -o tsv"
         } catch (e) {
         } // do nothing, record not found
-        if (!aRecordSet) {
-          this.az.az "network private-dns record-set a create -g ${resourceGroup} -z ${zone} -n ${recordName} --ttl ${ttl} --subscription ${subscription}"
-          this.az.az "network private-dns record-set a add-record -g ${resourceGroup} -z ${zone} -n ${recordName} -a ${serviceIP} --subscription ${subscription}"
-        } else {
+        try {
+          cnameRecordSet = this.az.az "network private-dns record-set cname show -g ${resourceGroup} -z ${zone} -n ${recordName} --subscription ${subscription} -o tsv"
```

Comment:
if the a recordset is found there's no point also checking for a cname.

Reply:
Without the check, if the A record doesn't exist and the CNAME record does, the pipeline would try to create an A record and it would fail since you can only have one or the other.

Reply:
you are checking if the a record exists first, if A exists you don't need to check for CNAME, which will save time, its important to keep this optimised as it runs a lot

Reply:
(unsure if you're still working on it but it's still not right, we should only do this check if the a record is missing)

Reply:
I get you now. I've moved the check for cname under the if statement.